test/unit/com/palantir/cassandra/db/MutationVerbHandlerTest.java

```diff
@@ -0,0 +1,135 @@
+import org.apache.cassandra.dht.ByteOrderedPartitioner;
+import org.apache.cassandra.exceptions.ConfigurationException;
+import org.apache.cassandra.locator.IEndpointSnitch;
+import org.apache.cassandra.locator.NetworkTopologyStrategy;
+import org.apache.cassandra.locator.PropertyFileSnitch;
+import org.apache.cassandra.locator.TokenMetadata;
+import org.apache.cassandra.net.MessageIn;
+import org.apache.cassandra.net.MessagingService;
+import org.apache.cassandra.service.StorageService;
+
+import static org.apache.cassandra.SchemaLoader.standardCFMD;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+@RunWith(OrderedJUnit4ClassRunner.class)
+public class MutationVerbHandlerTest
+{
+    private static final String KEYSPACE = "Keyspace1";
+    private static final String TABLE = "Standard1";
+    private static final MutationVerbHandler HANDLER = new MutationVerbHandler();
+
+    @BeforeClass
+    public static void setUp() throws ConfigurationException
+    {
+        DatabaseDescriptor.setDaemonInitialized();
+        IEndpointSnitch snitch = new PropertyFileSnitch();
+        DatabaseDescriptor.setEndpointSnitch(snitch);
+        Keyspace.setInitialized();
+    }
+
+    @Before
+    public void before() throws UnknownHostException
+    {
+        System.setProperty("palantir_cassandra.verify_keys_on_write", "true");
+
+        TokenMetadata metadata = StorageService.instance.getTokenMetadata();
+        metadata.clearUnsafe();
+        // DC1
+        metadata.updateNormalToken(ByteOrderedPartitioner.instance.getTokenFactory().fromString("A"),
+                                   InetAddress.getByName("127.0.0.1"));
+        metadata.updateNormalToken(ByteOrderedPartitioner.instance.getTokenFactory().fromString("C"),
+                                   InetAddress.getByName("127.0.0.2"));
+        // DC2
+        metadata.updateNormalToken(ByteOrderedPartitioner.instance.getTokenFactory().fromString("B"),
+                                   InetAddress.getByName("127.0.0.4"));
+        metadata.updateNormalToken(ByteOrderedPartitioner.instance.getTokenFactory().fromString("D"),
+                                   InetAddress.getByName("127.0.0.5"));
+
+        Keyspace.clear(KEYSPACE);
+        KSMetaData meta = KSMetaData.newKeyspace(KEYSPACE,
+                                                 NetworkTopologyStrategy.class,
+                                                 ImmutableMap.of("DC1", "1", "DC2", "1"),
+                                                 false,
+                                                 Collections.singleton(standardCFMD(KEYSPACE, TABLE)));
+        Schema.instance.setKeyspaceDefinition(meta);
+    }
+
+
+    @Test
+    public void doVerb_appliesWriteWhenOwned() throws IOException
+    {
+        MessageIn<Mutation> message = getMutationMessageForKey((byte) 0xf0a);
```

Comment:
Could you add a short comment to specify which endpoint that key and why? Same below